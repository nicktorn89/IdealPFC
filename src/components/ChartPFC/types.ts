export interface ChartsProps {
  data: ChartData[];
  size?: number;
}

export type ChartData = {
  name: string;
  value: number;
  color: string;
};
