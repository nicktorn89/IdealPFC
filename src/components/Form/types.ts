import { MainState } from 'src/modules/Main/types';

export interface FormProps {
  values: MainState;
  onChangeField: (e: React.MouseEvent<HTMLInputElement>) => void;
}
