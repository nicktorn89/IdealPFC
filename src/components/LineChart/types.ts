export interface LineChartProps {
  data: LineChartData[];
}

export type LineChartData = {
  name: string;
  calories: number;
};
