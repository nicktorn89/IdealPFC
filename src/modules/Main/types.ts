export interface MainState {
  weight: number;
  height: number;
  age: number;
  desiredWeight?: number;

  [key: string]: number | undefined;
}

export interface MainProps {
}

export interface Nutrients {
  protein: number;
  fats: number;
  carbohydrates: number;
}
