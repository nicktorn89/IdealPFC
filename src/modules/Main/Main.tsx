import React from 'react';
import { MainState, MainProps } from './types';
import FormComponent from '../../components/Form';
import HeaderComponent from 'src/components/Header';
import ChartPFC from 'src/components/ChartPFC';
import { PieChartData } from 'src/components/ChartPFC/types';
import LineChart from 'src/components/LineChart';
import { LineChartData } from 'src/components/LineChart/types';

const dataMock: PieChartData[] = [
  {
    name: 'Белки',
    value: 200,
    color: '#FFFF4C',
  },
  {
    name: 'Жиры',
    value: 150,
    color: '#FF4C4C',
  },
  {
    name: 'Углеводы',
    value: 650,
    color: '#4C4CFF',
  },
];

const anotherMock: LineChartData[] = [
  {
    name: '1 неделя',
    calories: 2500,
  },
  {
    name: '2 неделя',
    calories: 2300,
  },
  {
    name: '3 неделя',
    calories: 2100,
  },
  {
    name: '4 неделя',
    calories: 1900,
  },
  {
    name: '5 неделя',
    calories: 1700,
  },
];

export class Main extends React.Component<MainProps, MainState> {
  public readonly state = {
    weight: 0,
    height: 0,
    age: 0,
  };

  public handleFieldChange = ({ target }: React.MouseEvent<HTMLInputElement>) => {
    const { name, value } = target as HTMLInputElement;

    if (isNaN(+value)) return;

    this.setState({ [name]: +value });
  }

  public render = () => {
    return (
      <React.Fragment>
        <HeaderComponent />
        <FormComponent
          values={this.state}
          onChangeField={this.handleFieldChange}
        />
        <ChartPFC data={dataMock} />
        <LineChart data={anotherMock} />
      </React.Fragment>
    );
  }
}

export default Main;
