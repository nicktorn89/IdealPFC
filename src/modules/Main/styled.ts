import styled from 'styled-components';
import { media } from 'src/theme/mixin';
