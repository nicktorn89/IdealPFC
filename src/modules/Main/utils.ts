import { Nutrients } from './types';

export const roundToDecimal = (value: number) => Math.round(value * 10) / 10;

export const countNutrients = (height: number, weight: number, hasSportLoad?: boolean): Nutrients => 
  ({
    protein: roundToDecimal(weight * (hasSportLoad ? 2.5 : 1.5)),
    fats: roundToDecimal(weight * (hasSportLoad ? 1.5 : 0.8)),
    carbohydrates: roundToDecimal(weight * (hasSportLoad ? 4 : 2)),
  });
