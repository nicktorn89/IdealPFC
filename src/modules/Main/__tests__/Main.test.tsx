import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Main } from '../Main';
import { MainState } from '../types';

describe('<Main />', () => {
  const MainComponent = shallow(<Main />);
  const MainComponentToJson = shallowToJson(MainComponent);
  
  it('renders Main component', () => {
    expect(MainComponent).toExist();
  });

  it('should have 0 as default value in state', () => {
    const { weight, height, age } = MainComponent.state() as MainState;
    
    expect(weight).toBe(0);
    expect(height).toBe(0);
    expect(age).toBe(0);
  });

  it('desiredWeight should be undefined by default', () => {
    expect(MainComponent.state('desiredWeight')).toBeUndefined();
  });

  it('should be equal to snapshot', () => {
    expect(MainComponentToJson).toMatchSnapshot();
  });
});
