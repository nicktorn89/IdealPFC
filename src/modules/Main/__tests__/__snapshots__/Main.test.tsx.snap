// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should be equal to snapshot 1`] = `
<Fragment>
  <HeaderComponent />
  <FormComponent
    onChangeField={[Function]}
    values={
      Object {
        "age": 0,
        "height": 0,
        "weight": 0,
      }
    }
  />
  <ChartPFC
    data={
      Array [
        Object {
          "color": "#FFFF4C",
          "name": "Белки",
          "value": 200,
        },
        Object {
          "color": "#FF4C4C",
          "name": "Жиры",
          "value": 150,
        },
        Object {
          "color": "#4C4CFF",
          "name": "Углеводы",
          "value": 650,
        },
      ]
    }
  />
  <LineChart
    data={
      Array [
        Object {
          "calories": 2500,
          "name": "1 неделя",
        },
        Object {
          "calories": 2300,
          "name": "2 неделя",
        },
        Object {
          "calories": 2100,
          "name": "3 неделя",
        },
        Object {
          "calories": 1900,
          "name": "4 неделя",
        },
        Object {
          "calories": 1700,
          "name": "5 неделя",
        },
      ]
    }
  />
</Fragment>
`;
